python/appledoublekaitai/kaitai_utils.py

```diff
@@ -1,62 +1,200 @@
+        @param path: string representing the jsonpath to the current node in the object tree
+        @return: dictionary containing parsed fields and their respective parsed values in a dictionary
+        """
+        offsets = None
+        parameters_dict = _parameters_dict(instance)
+        if '_debug' in parameters_dict:
+            offsets = parameters_dict['_debug']
+        for key, value_object in parameters_dict.items():
+            if not _is_public_property(key, value_object):
+                continue
             if _is_kaitai_struct(value_object):
-                yield _to_lower_camel_case(key), _object_to_dict(value_object)
+                yield _to_lower_camel_case(key), self._object_to_dict(value_object, path + '.' + key)
             elif _is_list(value_object):
-                yield _to_lower_camel_case(key), _list_to_dict(value_object)
+                yield _to_lower_camel_case(key), self._list_to_dict(value_object, path)
+            elif isinstance(value_object, bytes):
+                yield self._process_bytes(value_object, offsets or {}, path, key)
             else:
                 yield _to_lower_camel_case(key), _process_value(value_object)
 
+    def _process_bytes(self, value_object: Any, offsets: dict, path: str, key: str) -> tuple[str, Any]:
+        """
+        Method responsible for converting a byte array from the Kaitai object tree to a trace if necessary and / or putting
+        it in the resulting JSON output.
+
+        @param value_object: part of the Kaitai object tree that represents a byte array
+        @param offsets: the offsets at which the byte array was read in the original inputData
+        @param path: JSONpath to this object in the resulting JSON
+        @param key: key at which value_object was stored in the Kaitai object tree
+        """
+        if len(value_object) > self.max_byte_array_length:
+            if len(value_object) < hansken_extraction_plugin.runtime.constants.MAX_CHUNK_SIZE:
+                length = offsets[key]['end'] - offsets[key]['start']
+                child_builder = self.trace.child_builder(path)
+                if self.has_process and offsets is not None:
+                    child_builder.update(data={'raw': value_object}).build()
+                    return _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
+                else:
+                    child_builder.add_transformation('raw', RangedTransformation([Range(offsets[key]['start'], length)]))
+                    child_builder.build()
+                    return _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
+            else:
+                return _to_lower_camel_case(key), f'data block of size: {len(value_object)} (not added as child trace because size exceeds MAX_CHUNK_SIZE)'
+        else:
+            return _to_lower_camel_case(key), value_object.hex()
+
+
```

Comment:
Remove one empty line here.

Reply:
Again, this should be fixed in the template (separate PR).